doc/examples.md

```diff
@@ -0,0 +1,196 @@
+
+## Users and groups
+
+```yaml
+passwd:
+  users:
+    - name: core
+      password_hash: password1
+      ssh_authorized_keys:
+        - key1
+```
+
+This example modifies the existing `core` user, giving it a known password hash
+(this will enable login via password), and setting its ssh key.
+
+```yaml
+passwd:
+  users:
+    - name: user1
+      password_hash: password1
+      ssh_authorized_keys:
+        - key1
+        - key2
+    - name: user2
+      ssh_authorized_keys:
+        - key3
+```
+
+This example will create two users, `user1` and `user2`. The first user has a
+password set and two ssh public keys authorized to log in as the user. The
+second user doesn't have a password set (so log in via password will be
+disabled), and one ssh key.
+
+```yaml
+passwd:
+  users:
+    - name: user1
+      password_hash: password1
+      ssh_authorized_keys:
+        - key1
+      create:
+        home_dir: /home/user1
+        no_create_home: true
+        groups:
+          - wheel
+          - plugdev
+        shell: /bin/zsh
+```
+
+This example creates one user, `user1`, with the password hash `password1`, and
+sets up one ssh public key for the user. The user is also given the home
+directory `/home/user1`, but it's not created, the user is added to the `wheel`
+and `plugdev` groups, and the user's shell is set to `/bin/zsh`.
+
+## Storage and Files
+
+### Files
+
+```yaml
+storage:
+  files:
+    - path: /opt/file1
+      filesystem: root
+      contents:
+        inline: Hello, world!
+      mode: 0644
+      user:
+        id: 500
+      group:
+        id: 501
+```
+
+This example creates a file at `/opt/file` with the contents `Hello, world!`,
+permissions 644 (so readable and writable by the owner, and only readable by
+everyone else), and the file is owned by user uid 500 and gid 501.
+
+```yaml
+storage:
+  files:
+    - path: /opt/file2
+      filesystem: root
+      contents:
+        remote:
+          url: http://example.com/file2
+          compression: gzip
+          verification:
+            hash:
+              function: sha512
+              sum: 4ee6a9d20cc0e6c7ee187daffa6822bdef7f4cebe109eff44b235f97e45dc3d7a5bb932efc841192e46618f48a6f4f5bc0d15fd74b1038abf46bf4b4fd409f2e
+      mode: 0644
+```
+
+This example fetches a gzip-compressed file from `http://example.com/file2`,
+makes sure that it matches the provided sha512 hash, and writes it to
+`/opt/file2`.
+
+### Filesystems
+
+```yaml
+storage:
+  filesystems:
+    - name: filesystem1
+      mount:
+        device: /dev/disk/by-partlabel/ROOT
+        format: btrfs
+        create:
+          force: true
+          options:
+            - -L
+            - ROOT
+```
+
+This example formats the root filesystem to be `btrfs`, and names it
+`filesystem1` (primarily for use in the `files` section).
+
+## Systemd Units
+
+```yaml
+systemd:
+  units:
+    - name: etcd-member.service
+      dropins:
+        - name: conf1.conf
+          contents: |-
```

Comment:
> the final line break and any trailing empty lines are excluded from the scalar’s content

Reference if you want to read more on it: http://www.yaml.org/spec/1.2/spec.html#id2794534

I'm indifferent to the syntax used here, and can change it to `|` if you'd prefer.